eng/pipelines/templates/scripts/merge-base-branch-to-target.ps1

```diff
@@ -0,0 +1,48 @@
+<#
+.SYNOPSIS
+
+.DESCRIPTION
+Merge the base branch to the target branch. E.g merge main to the top of the live
+
+.PARAMETER BaseBranch
+The base branch where we want to merge changes from. E.g. main branch in docs repo
+
+.PARAMETER TargetBranch
+The target branch where we want to merge changes to. E.g. live branch in docs repo
+
+#>
+param(
+  [Parameter(mandatory=$true)]
+  [string] $BaseBranch,
+  [Parameter(mandatory=$true)]
+  [string] $TargetBranch
+)
+
+# Switching working branch to target branch, e.g. live
+$currentBranch = (git branch --show-current)
+Write-Host "Current branch: $currentBranch, Base branch: $TargetBranch."
+if ($TargetBranch -ne $currentBranch) {
+    git show-ref --verify --quiet refs/heads/$TargetBranch
+    if ($LASTEXITCODE -eq 0) {
+        Write-Host "git checkout $TargetBranch"
+        git checkout $TargetBranch
+    }
+    else {
+        Write-Host "git checkout -b $TargetBranch -t origin/$TargetBranch"
+        git checkout -b $TargetBranch -t origin/$TargetBranch
+    }
+}
+
+# Always print out the commit SHA
+$commitSHA = git log -1 --format=tformat:%H 
+Write-Host "The $TargetBranch last commit SHA before merge: $commitSHA"
+
+# Merge base branch to target
+Write-Host "git -c user.name=`"azure-sdk`" -c user.email=`"azuresdk@microsoft.com`" merge $BaseBranch"
+git -c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com" merge $BaseBranch > $null
```

Comment:
Why pipe to null? This information might be interesting for debugging any issues.

Reply:
We should be checking error codes for these calls that could fail so that we can correctly fail if they fail.

Reply:
The merge info includes thousands of lines.(The main to live usually includes huge changes) It is a little annoying to me.

Reply:
Does error message deliver more information than error code?
Why we add overheads to check the error code?

We only dump the output message for success case. We will show error messages here and fail the script.

Reply:
Checking the error code allows us to correct handle the error and skip the next steps in the script as needed. As for the log I agree it might be long but as it stands now we will not get any useful information if there is a failure as we don't capture or display any of the output from the merge command.

Reply:
We could dump the output to a `.log` file and upload an artifact for later inspection. 

I think we should also check the `$LASTEXITCODE` here because we could detect merge problems long before getting to the push phase. 

For example: this task succeeded but the SHA didn't change and the push task failed: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1814857&view=logs&j=ea44f861-79e0-56b9-d4b2-32eb1e9d33c1&t=fe522bad-2aee-5cd6-aca5-f98a7ce68864&l=19

Reply:
Think about to change back to live branch replacement approach.

Reply:
The error thrown by git is not trappable by PS.
We have to use try catch for the failure cases.

Reply:
No such steps anymore. The git error indeed use exit code. I tested in bash script. My bad